kyo-core/shared/src/main/scala/kyo/package.scala

```diff
@@ -72,19 +72,19 @@ package object kyo:
         inline def failTag[T, U](
             inline actual: Tag[U]
         ): Nothing =
-            bug(s"Unexpected effect '${actual.parse}' found in 'pure'.")
+            bug(s"Unexpected effect '${actual.show}' found in 'pure'.")
 
-        inline def failTag[T, U](
-            inline actual: Tag[U],
-            inline expected: Tag[T]
+        inline def failTag(
+            inline actual: Tag[?],
+            inline expected: Tag[?]*
```

Comment:
Since the new `Tag` doesn't support intersection types, I'm simulating them here for error reporting via a vararg param.